File: Python3/minimum_index_sum_of_two_lists.py
class Solution:
    def findRestaurant(self, list1: List[str], list2: List[str]) -> List[str]:
        result = []
        least = len(list1) + len(list2)
        
        for i in range(len(list1)):
            for j in range(len(list2)):
                if list1[i] == list2[j] and (i+j) == least:
                    result = [list1[i]] + result
                if list1[i] == list2[j] and (i+j) < least: 
                    result = [list1[i]]
                    least = i + j
                
        return result

File: C++/minimum_index_sum_of_two_lists.cpp
class Solution {
public:
    vector<string> findRestaurant(vector<string>& list1, vector<string>& list2) {
        vector<string> result;
        int least = list1.size() + list2.size();
        
        for(int i = 0; i < list1.size(); i++) {
            for(int j = 0; j < list2.size(); j++) {
                if(list1[i] == list2[j] && (i+j) == least)
                    result.push_back(list1[i]);
                if(list1[i] == list2[j] && (i+j) < least) {
                    result.clear();
                    result.push_back(list1[i]);
                    least = i + j;
                }
            }
        }
                
        return result;
    }
};
